digdag-core/src/main/java/io/digdag/core/workflow/WorkflowExecutor.java

```diff
@@ -539,6 +540,18 @@ public void runWhile(BooleanSupplier cond)
         }
     }
 
+    protected <R> R catching(Supplier<R> func, R defaultValue, String errorMessag)
+    {
+        try {
+            return func.get();
+        }
+        catch (Exception e) {
+            logger.warn(errorMessag);
```

Comment:
we'd better to collect metrics for that.

Reply:
Thanks.  I added it.